### ISSUE TITLE
Redis warning issues

### ISSUE THREAD

**Issue by hyperknot** [phusion/passenger-docker#142]
On 0.9.18 / Ubuntu 14.04 / Redis 2.8 this warning was in Redis log:

> WARNING overcommit_memory is set to 0! Background save may fail under low memory condition. To fix this issue add 'vm.overcommit_memory = 1' to /etc/sysctl.conf and then reboot or run the command 'sysctl vm.overcommit_memory=1' for this to take effect.

0.9.19 / Ubuntu 16.04 / Redis 3.0 introduced two more warnings (in addition to the first one):

> WARNING: The TCP backlog setting of 511 cannot be enforced because /proc/sys/net/core/somaxconn is set to the lower value of 128.

and

> WARNING you have Transparent Huge Pages (THP) support enabled in your kernel. This will create latency and memory usage issues with Redis. To fix this issue run the command 'echo never > /sys/kernel/mm/transparent_hugepage/enabled' as root, and add it to your /etc/rc.local in order to retain the setting after a reboot. Redis must be restarted after THP is disabled.

For the last one, I tried: `echo never > /sys/kernel/mm/transparent_hugepage/enabled` however it terminated in `bash: /sys/kernel/mm/transparent_hugepage/enabled: Read-only file system`

**Comment by CamJN** [comment]
These issues must be resolved in the HOST OS, not the container. Remember you are sharing the same kernel using docker.

**Comment by hyperknot** [comment]
Oh, just for reference, here is how to fix all the issues as of Docker 1.12+:

1. sysctl:

```
net.core.somaxconn = 65535
vm.overcommit_memory = 1
```

2. `docker run` gets the following argument: 

```
--sysctl net.core.somaxconn=65535
```

3. this goes in `/etc/rc.local`: 

```
echo never > /sys/kernel/mm/transparent_hugepage/enabled
```
